controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -46,31 +53,32 @@ func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCI
 			return err
 		}
 	}
-
 	return nil
 }
 
 func (r *DSCInitializationReconciler) removeServiceMesh(instance *dsciv1.DSCInitialization) error {
 	// on condition of Managed, do not handle Removed when set to Removed it trigger DSCI reconcile to clean up
-	if instance.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
-		capabilities := []*feature.HandlerWithReporter[*dsciv1.DSCInitialization]{
-			r.serviceMeshCapability(instance, serviceMeshCondition(status.RemovedReason, "Service Mesh removed")),
-		}
+	if instance.Spec.ServiceMesh != nil {
```

Comment:
Can't we make a guard ifs instead of nesting across the whole change? That might be a bit easier to follow.

```suggestion
	if instance.Spec.ServiceMesh == nil {
		return nil
	}	
```